test/cluster_resource_test.go

```diff
@@ -0,0 +1,245 @@
+// +build e2e
+
+/*
+Copyright 2018 Knative Authors LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
+	buildv1alpha1 "github.com/knative/build/pkg/apis/build/v1alpha1"
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	knativetest "github.com/knative/pkg/test"
+	"github.com/knative/pkg/test/logging"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestClusterResource(t *testing.T) {
```

Comment:
i feel like we already have a lot of integration tests - what if we had the helm integration test use a cluster resource instead, then we could just have the one test?

Reply:
I feel this test is specific to the creation of the kubeconfig file, I would prefer to remove the kaniko test instead, WDYT?

Reply:
If the concern is only about then we could always consider refactoring tests to be parallel since they all e2e tests are self contained in their namespaces.

Reply:
my concern is less about time and more about:

1. Maintaining the integration tests
2. Relying on integration tests for coverage

this is definitely a philosophical point of view, but i recently watched [integration tests are a scam](https://blog.thecodewhisperer.com/permalink/integrated-tests-are-a-scam) and i realized that relying too much on integration tests can cause us to be lax with our unit test coverage, which in turn promotes bad design choices.

This is besides the fact that integration tests are slow and when they fail, it's very hard to know what to do.

(Watch this space for my upcoming kubecon talk which is pretty much all about this!!)

So I would like to minimize the surface of what we test in our integration tests as much as possible, and focus instead of organizing and unit testing our code in such a way that we don't feel like we need to cover it in integration tests #the-possibly-unrealistic-ideal

> I feel this test is specific to the creation of the kubeconfig file, I would prefer to remove the kaniko test instead, WDYT?

in an ideal world what i would really like is just one integration test that includes as much functionality as we think is reasonable, e.g. it uses a kube cluster, has a service account, builds and pushes an image.

Anyway feel free to ignore this and go ahead with this test but you'll probably see a PR from me in the future where I try to move our integration tests toward this ideal - we can always continue the discussion there :D

Reply:
for this specific case, I can't really merge this integration test  checking for kubeconfig with the helm one because we would need a actual 2nd cluster to target by the helm deployment. The kaniko test we can delete, or maybe the hello-world?

Reply:
> I can't really merge this integration test checking for kubeconfig with the helm one because we would need a actual 2nd cluster to target by the helm deployment. 

oh i see, i didnt realize that the integration test wasn't actually _using_ the kubeconfig

would you mind making a follow-up issue in our backlog that we should consider having an integration test that actually uses a second cluster?

alternatively:
we could use the cluster resource to (somehow) point at the current cluster, but maybe a different namespace or something?